### PR TITLE
BCN - Lucas Affini / Clara Ameller

### PR DESCRIPTION
**Pull request by lacazeto**
@claraameller

### CODE REVIEW

File: starter-code/src/viking.js [hunk 1]
```diff
@@ -1,11 +1,98 @@
 // Soldier
-function Soldier () {}
+function Soldier (health, strength) {
+    this.health = health;
+    this.strength = strength;
+
+    this.attack = function(){
```
Comment (tawebbcn) [inline starter-code/src/viking.js]: Don't define the object's methods inside the constructor. Instead, outside it, do:
```
Soldier.prototype.attack = function () {
  return this.strength;
};
```

File: starter-code/src/viking.js [hunk 2]
```diff
@@ -1,11 +1,98 @@
...
+        return this.strength;
+    };
+
+    this.receiveDamage = function(damage) {
```
Comment (tawebbcn) [inline starter-code/src/viking.js]: Same as before:
```
Soldier.prototype.receiveDamage = function (damage) {
  this.health -= damage;
};
```

File: starter-code/src/viking.js [hunk 3]
```diff
@@ -1,11 +1,98 @@
...
+        this.health -= damage;
+    }
+}
 
 // Viking
-function Viking () {}
+function Viking (name, health, strength) {
```
Comment (tawebbcn) [inline starter-code/src/viking.js]: Same problem, methods better outside the constructor.

File: starter-code/src/viking.js [hunk 4]
```diff
@@ -1,11 +1,98 @@
...
+            return this.name + " has died in act of combat";
+        }
+    }
+
+    this.battleCry = function(){
+        return "Odin Owns You All!";
+    }
+}
+
+Viking.prototype = Object.create(Soldier.prototype);
```
Comment (tawebbcn) [inline starter-code/src/viking.js]: After that, for proper inheritance's implementation, you need to add:
```
Viking.prototype.constructor = Viking;
```

File: starter-code/src/viking.js [hunk 5]
```diff
@@ -1,11 +1,98 @@
...
 
 // Saxon
-function Saxon () {}
+function Saxon (health, strength) {
```
Comment (tawebbcn) [inline starter-code/src/viking.js]: Same problems as in Viking.

File: starter-code/src/viking.js [hunk 6]
```diff
@@ -1,11 +1,98 @@
...
+            return "A Saxon has died in act of combat";
+        }
+    }
+}
+
+Saxon.prototype = Object.create(Soldier.prototype);
 
 // War
-function War () {}
+function War () {
```
Comment (tawebbcn) [inline starter-code/src/viking.js]: One more time, methods outside the constructor.

File: starter-code/src/viking.js [hunk 7]
```diff
@@ -1,11 +1,98 @@
...
+        var theVictm = Math.floor(Math.random() * saxonArmy.length); 
+        var isDead = saxonArmy[theVictm].receiveDamage(300);
+        if (isDead = "A Saxon has died in act of combat"){
+            this.saxonArmy.splice(theVictm,1);
+        }
+
+        return isDead;
+    }
+
+    this.saxonAttack = function(){
```
Comment (tawebbcn) [inline starter-code/src/viking.js]: This code looks exactly the same as vikingAttack(). Maybe you can use a function...

File: starter-code/src/viking.js (hunk 1 above)
Comment (lacazeto) [reply starter-code/src/viking.js]: Why is that? 
Wouldn´t this make more susceptible to error-prone?

Comment (tawebbcn) [reply starter-code/src/viking.js]: Mainly for performance and economy. We talked about this when you were doing the exercise. Of course, we can discuss it again. Anyways, here you have more info: https://stackoverflow.com/questions/4508313/advantages-of-using-prototype-vs-defining-methods-straight-in-the-constructor

**Pull request closed without merging**